### PR TITLE
[FLINK-26210][connector/pulsar] Fix the jaxb-api missing for Pulsar connector on Java 11

### PR DESCRIPTION
**Pull request by syhily**
## What is the purpose of the change

We exclude the jaxb-api in `pulsar-client-all` after upgrading to latest Pulsar. This would cause the class not found issue on Java 11. The JAXB APIs are considered to be Java EE APIs and therefore are no longer contained on the default classpath in Java SE 9. In Java 11, they are completely removed from the JDK.

## Brief change log

  - Add the jaxb-api dependency.
  - Disable the `PulsarSourceOrderedE2ECase` test on Java 11 for direct buffer memory error.

## Verifying this change

This change is already covered by existing tests, such as `PulsarSourceUnorderedE2ECase`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-end-to-end-tests/flink-end-to-end-tests-pulsar/src/test/java/org/apache/flink/tests/util/pulsar/PulsarSourceOrderedE2ECase.java [hunk 1]
```diff
@@ -29,11 +29,15 @@
...
 import org.apache.flink.tests.util.pulsar.common.PulsarContainerTestEnvironment;
+import org.apache.flink.testutils.junit.FailsOnJava11;
+
+import org.junit.experimental.categories.Category;
 
 /**
  * Pulsar E2E test based on connector testing framework. It's used for Failover & Exclusive
  * subscription.
  */
+@Category(value = {FailsOnJava11.class})
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-pulsar/src/test/java/org/apache/flink/tests/util/pulsar/PulsarSourceOrderedE2ECase.java:40]: Why not just add a dependency on jaxb-api in the Java 11 profile?...

Actually, why is this necessary at all if we copy the jaxb-api jar in the pom?

File: flink-connectors/flink-sql-connector-pulsar/pom.xml [hunk 2]
```diff
@@ -69,6 +69,7 @@ under the License.
 									<include>org.bouncycastle:bcprov-jdk15on</include>
 									<include>org.bouncycastle:bcutil-jdk15on</include>
 									<include>org.slf4j:jul-to-slf4j</include>
+									<include>javax.xml.bind:jaxb-api</include>
```
Comment (zentol) [inline flink-connectors/flink-sql-connector-pulsar/pom.xml:72]: FYI: flink-dist bundles jaxb-api

**Review by zentol — CHANGES_REQUESTED** [review]

File: flink-connectors/flink-sql-connector-pulsar/pom.xml (hunk 2 above)
Comment (zentol) [reply flink-connectors/flink-sql-connector-pulsar/pom.xml:72]: (so it shouldn't be necessary for connectors to bundle it as well)

Comment (zentol) [reply flink-connectors/flink-sql-connector-pulsar/pom.xml:72]: And _if_ you bundle it it should be done in such a way that it is only loaded from the jar on Java 11, as it is done for flink-dist.

Comment (syhily) [reply flink-connectors/flink-sql-connector-pulsar/pom.xml:72]: > FYI: flink-dist bundles jaxb-api already

If flink-dist bundles the jaxb-api. Why we would meet a class not found error on flink ci?
https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=31702&view=logs&j=6e8542d7-de38-5a33-4aca-458d6c87066d&t=5846934b-7a4f-545b-e5b0-eb4d8bda32e1&l=15537

Comment (zentol) [reply flink-connectors/flink-sql-connector-pulsar/pom.xml:72]: Presumably because it is missing outside of the container, i.e., by the pulsar environment.

File: flink-end-to-end-tests/flink-end-to-end-tests-pulsar/src/test/java/org/apache/flink/tests/util/pulsar/PulsarSourceOrderedE2ECase.java (hunk 1 above)
Comment (zentol) [reply flink-end-to-end-tests/flink-end-to-end-tests-pulsar/src/test/java/org/apache/flink/tests/util/pulsar/PulsarSourceOrderedE2ECase.java:40]: Is there a ticket for that?

Comment (zentol) [reply flink-end-to-end-tests/flink-end-to-end-tests-pulsar/src/test/java/org/apache/flink/tests/util/pulsar/PulsarSourceOrderedE2ECase.java:40]: Does the connector as a whole not work on Java 11, or is it an issue with the pulsar test classes?

File: flink-connectors/flink-sql-connector-pulsar/pom.xml (hunk 2 above)
Comment (syhily) [reply flink-connectors/flink-sql-connector-pulsar/pom.xml]: Yeah, I add the `jaxb-api` only for e2e tests now.

File: flink-end-to-end-tests/flink-end-to-end-tests-pulsar/pom.xml [hunk 3]
```diff
@@ -33,6 +33,7 @@ under the License.
 	<properties>
 		<pulsar.version>2.9.1</pulsar.version>
 		<bouncycastle.version>1.69</bouncycastle.version>
+		<jaxb-api.version>2.3.1</jaxb-api.version>
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-pulsar/pom.xml:36]: You could just use the `jaxb.api.version` property from the root pom.
